moto/batch/models.py

```diff
@@ -198,14 +200,32 @@ def __init__(
         self.container_properties = container_properties
         self.arn = None
         self.status = "ACTIVE"
-
+        self.tagger = TaggingService()
         if parameters is None:
             parameters = {}
         self.parameters = parameters
 
         self._validate()
         self._update_arn()
 
+        tags = self._format_tags(tags)
+        # Validate the tags before proceeding.
+        errmsg = self.tagger.validate_tags(tags or [])
+        if errmsg:
+            raise ValidationError(errmsg)
+
+        self.tagger.tag_resource(self.arn, tags or [])
+
+    def _format_tags(self, tags):
+        formatted_tags = []
+        for item in tags.items():
+            tag = {}
+            tag["Key"] = item[0]
+            tag["Value"] = item[1]
+            print(tag)
+            formatted_tags.append(tag)
+        return formatted_tags
```

Comment:
```suggestion
        return [{"Key": k, "Value": v} for k, v in tags.items()]
```
This is more readable, IMO - and at the very least gets rid of the print-statement :)